### PR TITLE
Update enzyme recipe to work with latest enzyme

### PR DESCRIPTION
**Pull request by jonathansamines**
React's enzyme recipe was updated to work with latest react (v16) and enzyme (v2) versions. I made the following assumptions:

- React is already configured in the project, therefore no instructions to add `react`, `react-dom` and `prop-types` packages were added
- Used the [browser testing recipe](https://github.com/avajs/ava/blob/master/docs/recipes/browser-testing.md) structure to guide the user through the configuration process.

Let me know if additional changes are required.

Solves #2522

### CODE REVIEW

File: docs/recipes/react.md [hunk 1]
```diff
@@ -26,22 +26,58 @@ You can find more information in [`@ava/babel`](https://github.com/avajs/babel).
...
 
-Let's first see how to use AVA with one of the most popular React testing libraries: [Enzyme](https://github.com/airbnb/enzyme).
+Let's first see how to use AVA with one of the most popular React testing libraries: [Enzyme](https://github.com/enzymejs/enzyme).
 
 If you intend to only use [shallow component rendering](https://facebook.github.io/react/docs/test-utils.html#shallow-rendering), you don't need any extra setup.
 
-First install [Enzyme required packages](https://github.com/airbnb/enzyme/#installation):
+### Install enzyme
+
+First install [Enzyme required packages](https://github.com/enzymejs/enzyme#installation):
```
Comment (novemberborn) [inline docs/recipes/react.md:35]:
```suggestion
First install [Enzyme and its required adapter](https://github.com/enzymejs/enzyme#installation):
```

File: docs/recipes/react.md [hunk 2]
```diff
@@ -26,22 +26,58 @@ You can find more information in [`@ava/babel`](https://github.com/avajs/babel).
...
 
 ```console
-$ npm install --save-dev enzyme react-addons-test-utils react-dom
+$ npm install --save-dev enzyme enzyme-adapter-react-16
 ```
 
+### Setup enzyme
```
Comment (novemberborn) [inline docs/recipes/react.md:41]:
```suggestion
### Set up Enzyme
```

**Review by novemberborn — COMMENTED** [review]
Oh, thanks! I've suggested some minor changes, what do you reckon?

File: docs/recipes/react.md [hunk 3]
```diff
@@ -26,22 +26,58 @@ You can find more information in [`@ava/babel`](https://github.com/avajs/babel).
...
 Then you can use Enzyme straight away:
 
+`test.js`:
+
 ```js
 const test = require('ava');
 const React = require('react');
-const {shallow} = require('enzyme');
+const PropTypes = require('prop-types');
+const { shallow } = require('enzyme');
```
Comment (sindresorhus) [inline docs/recipes/react.md:80]:
```suggestion
const {shallow} = require('enzyme');
```

File: docs/recipes/react.md [hunk 4]
```diff
@@ -26,22 +26,58 @@ You can find more information in [`@ava/babel`](https://github.com/avajs/babel).
...
+
+Create a helper file, prefixed with an underscore. This ensures AVA does not treat it as a test.
+
+`test/_setup-enzyme-adapter.js`:
+
+```js
+const Enzyme = require('enzyme');
+const Adapter = require('enzyme-adapter-react-16');
+
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (sindresorhus) [inline docs/recipes/react.md:51]:
```suggestion
Enzyme.configure({
	adapter: new Adapter()
});
```

File: docs/recipes/react.md (hunk 3 above)
Comment (jonathansamines) [reply docs/recipes/react.md]: Updated destructuring code style

**Comment by novemberborn** [comment]
Thanks @jonathansamines!

**Pull request merged**